Newsletters: add 257 (2023-06-28) (#1164)
bitcoinops/bitcoinops.github.io: _includes/specials/policy/en/07-network-interest.md

> <!-- 300 to 1000 words put title in main newsletter put links in this file for
> any subheads use h3 (i.e., ###) illustrations welcome (max width 800px) if
> uncertain about anything, just do what seems best and harding will edit -->
> 

Reviewer: ```suggestion
```

for spending them. Even so, [“dust storms” with thousands of transactions
occurred as recently as 2020][lopp storms].

When it became popular to use bare multisig outputs to publish data onto the
blockchain, the definition of standard transactions was amended to permit a
single OP_RETURN output as an alternative. People realized that it would be
impossible to prevent users from publishing data on the blockchain, but at
least such data would not need to live in the UTXO set forever when published
in outputs that could never be spent. Bitcoin Core 0.13.0 introduced a start-up
option `-permitbaremultisig` that users may toggle to reject unconfirmed
transactions with bare multisig outputs.

While the consensus rules allow output scripts to be freeform, only a few
well-understood patterns are relayed by Bitcoin Core nodes. This makes it
easier to reason about many concerns in the network, including validation costs
and protocol upgrade mechanisms. For example, an input script that contains
op-codes, a P2SH input with more than 15 signatures, or a P2WSH input whose
witness stack has more than 100 items each would make a transaction

> non-standard. (Check out this [policy overview][instagibbs policy zoo]for more

Reviewer: ```suggestion
non-standard. (Check out this [policy overview][instagibbs policy zoo] for more
```

central points of failure, network-wide software upgrades involve a coordinated
effort between tens of thousands of independent node operators. Nodes will not
relay transactions that make use of any reserved upgrade hooks until their
meaning has been defined. This discouragement is meant to dissuade applications
from independently creating conflicting standards, which would make it
impossible to adopt one application’s standard into consensus without
invalidating another’s. Also, when a consensus change does happen, nodes that
do not upgrade immediately—and thus do not know the new consensus rules—cannot
be “tricked” into accepting a now-invalid transaction into their mempools. The
proactive discouragement helps nodes be forward-compatible and enables the
network to safely upgrade consensus rules without requiring a completely
synchronized software update.

We can see that using policy to protect shared network resources aids in
protecting the network’s characteristics, and keeps paths for future protocol
development open. Meanwhile, we are seeing how the friction of growing the
network against a strictly limited blockweight has been driving adoption of

> best practices, good technical design, and innovation: in next week’s post we
> will be exploring mempool policy as an interface for second layer protocols and

Reviewer: ```suggestion
best practices, good technical design, and innovation: next week’s post
will explore mempool policy as an interface for second layer protocols and
```
